index.js

```diff
@@ -169,7 +169,7 @@ jasmine.Expectation.prototype.wrapCompare = function(name, matcherFactory) {
 
     if (!webdriver.promise.isPromise(expectation.actual) &&
         !webdriver.promise.isPromise(expected)) {
```

Comment:
Because once you call when(), you force the call to become async. With this check the it doesn't become async unless it needs to. This was the original reason this code changed, and broke, in the first place. See https://github.com/angular/protractor/issues/2894